libs/gltfio/src/AssetLoader.cpp

```diff
@@ -88,12 +90,38 @@ static constexpr cgltf_material kDefaultMat = {
     },
 };
 
-static const char* getNodeName(const cgltf_node* node, const char* defaultNodeName) {
-    if (node->name) return node->name;
-    if (node->mesh && node->mesh->name) return node->mesh->name;
-    if (node->light && node->light->name) return node->light->name;
-    if (node->camera && node->camera->name) return node->camera->name;
-    return defaultNodeName;
+static std::string getNodeName(cgltf_node const* node, char const* defaultNodeName) {
+    auto getNameImpl = [node, defaultNodeName]() -> char const* {
+        if (node->name) return node->name;
+        if (node->mesh && node->mesh->name) return node->mesh->name;
+        if (node->light && node->light->name) return node->light->name;
+        if (node->camera && node->camera->name) return node->camera->name;
+        return defaultNodeName;
+    };
+
+    std::string sImpl(getNameImpl());
+
+    // We handle the potential case of escaped characters in the JSON which should be properly
+    // interpreted as unicode. See spec:
+    // https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#json-encoding
+    std::string sEscapedUnicode;
+    size_t cur = 0, idx = 0;
+    std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> conv;
+    while ((idx = sImpl.find("\\u", cur)) != std::string::npos) {
+        auto endIdx = sImpl.find_first_not_of("0123456789abcdefABCDEF", idx + 2);
```

Comment:
I think we need to limit this to only 6 characters. Otherwise we could have input like this which causes a `std::range_error: wstring_convert: to_bytes error`:

```
\u6211\u53EBBenjamin
```

(should be 我叫Benjamin)

> If the character is in the Basic
   Multilingual Plane (U+0000 through U+FFFF), then it may be
   represented as a six-character sequence: a reverse solidus, followed
   by the lowercase letter u, followed by four hexadecimal digits that
   encode the character's code point.

See https://www.ietf.org/rfc/rfc4627.txt

Reply:
Thanks for the spec link.  This makes things a lot easier to understand.

Reply:
Addressed.

Reply:
LGTM. My other recommendation is that `to_bytes` can throw a `std::range_error` error if the unicode point is not valid, but not sure if we care about handling that case since that would mean the string is malformed.

Reply:
I feel that's more of the user's responsibility.  Let's discuss again if/when someone files a bug on it.